Makefile

```diff
@@ -0,0 +1,84 @@
+
+# Image URL to use all building/pushing image targets
+IMG ?= controller:latest
+# Produce CRDs that work back to Kubernetes 1.11 (no version conversion)
+CRD_OPTIONS ?= "crd:trivialVersions=true"
+
+# Get the currently used golang install path (in GOPATH/bin, unless GOBIN is set)
+ifeq (,$(shell go env GOBIN))
+GOBIN=$(shell go env GOPATH)/bin
+else
+GOBIN=$(shell go env GOBIN)
+endif
+
+all: manager
+
+# Run tests
+test: generate fmt vet manifests
+	go test ./... -coverprofile cover.out
+
+# Build manager binary
+manager: generate fmt vet
+	go build -o bin/manager main.go
+
+# Run against the configured Kubernetes cluster in ~/.kube/config
+run: generate fmt vet manifests
+	go run ./main.go
+
+# Install CRDs into a cluster
+install: manifests
+	kustomize build config/crd | kubectl apply -f -
```

Comment:
This kind of thing concerns me. What if the user assumes that `make install` installs the go binary locally, and ends up modifying their cluster instead.

Reply:
Maybe this kind of thing would be better in the readme.

Reply:
The makefile is autogenerated. It has really useful cases but they are poorly named. That's something we can easily change. Top ones to know:
`make manifests`: creates the CRD
`make install`: install the CRD in your kubeconfig cluster
`make docker-build IMG=controller:V0.0.1` Builds the docker image with specified tag
`make docker-push IMG=controller:V0.0.1` push specified image
`make deploy IMG=controller:V0.0.1` deploys the controller with specified image tag
`make` builds the go app
`make run` runs your go app using kubeconfig creds